src/flashbar/collapsible-flashbar.tsx

```diff
@@ -0,0 +1,392 @@
+
+  const getElementsToAnimate = useCallback(() => {
+    const flashElements = isFlashbarStackExpanded ? expandedItemRefs.current : collapsedItemRefs.current;
+    return { ...flashElements, toggleButton: toggleElementRef.current };
+  }, [isFlashbarStackExpanded]);
+
+  const prepareAnimations = useCallback(() => {
+    const rects = getDOMRects(getElementsToAnimate());
+    setInitialAnimationState(rects);
+  }, [getElementsToAnimate]);
+
+  const { baseProps, breakpoint, isReducedMotion, isVisualRefresh, mergedRef } = useFlashbar({
+    items,
+    ...restProps,
+    onItemsAdded: newItems => {
+      setEnteringItems([...enteringItems, ...newItems]);
+    },
+    onItemsChanged: options => {
+      // If not all items have ID, we can still animate collapse/expand transitions
+      // because we can rely on each item's index in the original array,
+      // but we can't do that when elements are added or removed, since the index changes.
+      if (options?.allItemsHaveId && !options?.isReducedMotion) {
+        prepareAnimations();
+      }
+    },
+    onItemsRemoved: removedItems => {
+      setExitingItems([...exitingItems, ...removedItems]);
+    },
+  });
+
+  const isFocusVisible = useFocusVisible();
+  const collapsedItemRefs = useRef<Record<string | number, HTMLElement | null>>({});
+  const expandedItemRefs = useRef<Record<string | number, HTMLElement | null>>({});
+  const [initialAnimationState, setInitialAnimationState] = useState<Record<string | number, DOMRect> | null>(null);
+  const listElementRef = useRef<HTMLUListElement | null>(null);
+  const toggleElementRef = useRef<HTMLDivElement | null>(null);
+  const [transitioning, setTransitioning] = useState(false);
+  const [listTop, setListTop] = useState<number>();
+
+  const flashbarElementId = useUniqueId('flashbar');
+  const itemCountElementId = useUniqueId('item-count');
+
+  useEffect(() => {
+    if (items.length <= maxNonCollapsibleItems) {
+      setIsFlashbarStackExpanded(false);
+    }
+  }, [items.length]);
+
+  const animateFlash = !isReducedMotion;
+
+  function toggleCollapseExpand() {
+    sendToggleMetric(items.length, !isFlashbarStackExpanded);
+    if (!isReducedMotion) {
+      prepareAnimations();
+    }
+    setIsFlashbarStackExpanded(prev => !prev);
+  }
+
+  const updateBottomSpacing = useCallback(() => {
+    // Allow vertical space between Flashbar and page bottom only when the Flashbar is reaching the end of the page,
+    // otherwise avoid spacing with eventual sticky elements below.
+    const listElement = listElementRef?.current;
+    const flashbar = listElement?.parentElement;
+    if (listElement && flashbar && listTop !== undefined) {
+      const bottom = listTop + listElement.clientHeight;
+      const windowHeight = window.innerHeight;
+      flashbar.style.paddingBottom = '';
+      if (isFlashbarStackExpanded && bottom + parseInt(getComputedStyle(flashbar).paddingBottom) <= windowHeight) {
+        flashbar.style.paddingBottom = '0';
```

Comment:
We can, but still using direct DOM manipulation (e.g, with `classList.add` and `classList.remove` because when expanding, we want this change to be already applied before rendering, so that the animation goes directly to the desired place instead of first animating and then adding the bottom spacing in the case that the Flashbar reaches the end of the page when it expands.

Reply:
> instead of first animating and then adding the bottom spacing

Well, to be exact this is not what would happen. It would render at once but the initial position of the animation would be a bit off.

Reply:
Thanks for adding the class!